contents/blog/how-come-we-ship-so-much.md

```diff
@@ -0,0 +1,121 @@
+
+* product analytics
+* session replay
+* feature flags
+* a/b testing
+* heatmaps
+* SDKs for all of the above
+* a basic CDP to stream data to warehouses
+* data warehouse MVP
+* a wildly extensive website
+* ... and those are things that have worked
+
+All of the above are being used a lot too - we're ingesting 10s of billions of events a month, have 10s of thousands of companies regularly using our service and over 100K user sign ups.
+
+One of the most common questions I get asked is how do we manage to build so much so quickly.
+
+## We designed the company around speed
+
+We provide an all-in-one platform to help engineers build better products. We have a list of 25+ products we want to build.
+
+That means we had to design our company primarily for speed.
+
+## These ideas may _suck_ for you
+
+Before you start making changes, think hard about what your product should be optimized for.
+
+For example, if you are shipping the latest iPhone, complete design control is _more_ important to your users than having all the features. Likewise, if you are building software and are building a tiny product that has better UX than an incumbent, design and control matters.
+
+I _suspect_ that you could still get higher quality with more autonomy - my approach would probably be having lots of designers and engineers collaborating with relatively few product managers, but I cannot speak to the above as well since my experience is building a multi product platform.
+
+## No product management by default
+
+The responsibilities of a product manager still exist, but our engineers (the concept internally is a "product engineer"):
+
+* decide what to build within their product (and they often suggest new products - but the decision on this is made by my cofounder and me)
+* talk to users
+* have complete access to metrics (especially for their own product - ie revenue, hosting cost, staff cost)
+
+We have one product manager who happily steps in reactively if a small team needs more support.
+
+The above _requires_ a lot of context setting from the company. We are _particularly transparent_ - sharing board slides, financial information, fundraising progress before, during and after it happens, and much more. See our public handbook for a lot of this stuff.
+
+## No design by default
+
+When we started, we felt that if product designers dictate to engineers, the best engineers would eventually quit as they'd want more freedom.
+
+As we grew, we realized that engineers _can_ design the UX of a product. Especially if we hire people that have this skillset.
+
+We have people that can help engineers with design, but they are very happy to work reactively and fast - and they don't block merging work during QA. This means we _do_ ship work where the design isn't perfect, but we won't drop things that are important and will keep improving them.
+
+For example, we've redesigned this entire website ~5 times.
```

Comment:
Might be worth saying that this doesn't mean "totally no design". We did invest quite a lot in getting our design system up and running, and worked with a designer to get it this far... Then we moved on to the "no design" phase. Had we been in this phase from the beginning, the app would likely now look like unicorn vomit. Because we have the design system in place, we can move fast an only focus on UX, which engineers are able to do.

Reply:
about to push update with this clarification